cmd/neofs-cli/modules/object/put.go

```diff
@@ -55,22 +56,43 @@ func initObjectPutCmd() {
 	flags.Bool(noProgressFlag, false, "Do not show progress bar")
 
 	flags.String(notificationFlag, "", "Object notification in the form of *epoch*:*topic*; '-' topic means using default")
+	flags.Bool(binaryFlag, false, "Deserialize object structure from given file.")
 }
 
 func putObject(cmd *cobra.Command, _ []string) {
+	binary, _ := cmd.Flags().GetBool(binaryFlag)
+	cidVal, _ := cmd.Flags().GetString(cidFlag)
+
+	if !binary && cidVal == "" {
+		common.ExitOnErr(cmd, "", fmt.Errorf("required flag \"%s\" not set", cidFlag))
+	}
 	pk := key.GetOrGenerate(cmd)
 
 	var ownerID user.ID
-	user.IDFromKey(&ownerID, pk.PublicKey)
-
 	var cnr cid.ID
-	readCID(cmd, &cnr)
 
-	filename := cmd.Flag("file").Value.String()
+	var payloadReader io.Reader
```

Comment:
can we unite the declaration with the definition of the `payloadReader` var?

Reply:
Refactored